source/extensions/original_ip_detection/custom_header/custom_header.cc

```diff
@@ -0,0 +1,44 @@
+#include "extensions/original_ip_detection/custom_header/custom_header.h"
+
+#include "common/network/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace OriginalIPDetection {
+namespace CustomHeader {
+
+CustomHeaderIPDetection::CustomHeaderIPDetection(
+    const envoy::extensions::original_ip_detection::custom_header::v3::CustomHeaderConfig& config)
+    : header_name_(config.header_name()),
+      allow_trusted_address_checks_(config.allow_extension_to_set_address_as_trusted()) {
+  if (config.has_reject_options()) {
+    const auto reject_code = toErrorCode(config.reject_options().status_on_error().code());
```

Comment:
sorry for being unclear, I was thinking rejecting the config if an invalid code were set.

I'd be fine doing something like the local reply extension which just proto validates
 google.protobuf.UInt32Value status_code = 2 [(validate.rules).uint32 = {lt: 600 gte: 200}];
though you can narrow to 4xx/5xx given I don't think the other codes make sense.